src/content/1.7/testing/web-acceptance-tests/how-to-contribute-and-create-web-acceptance-tests.md

```diff
@@ -0,0 +1,212 @@
+
+Our team uses POM as an architecture to organize the code: on one side we have our page logic, and on the other side we have our test logic.
+
+The goal is to write your test code once, and only change the page logic when someone updates the application.
+
+Both are described in the following paragraphs.
+
+### Pages
+
+#### Name
+The name of the class must be simple and linked to the page in the shop. You should be able to find the corresponding page in the application using the folder hierarchy, and the class name.
+
+The name of the file is a little different: the main page name is always `index.js`.
+
+Example:
+ 
+For BO products pages located at `pages/catalog/products/`, we use 2 classes:
+- `index.js` for products listing page (main page)
+- `add.js` for add/edit product page
+ 
+#### Inheritance
+We heavily use inheritance in the pages to make sure all our generic methods are available from everywhere, and you don’t need to instantiate separate objects. 
+
+There is a main mother class, called `CommonPage`, located at the root of the `pages` folder.
+
+We have another level of inheritance in each root folder (`BO` and `FO`) through the classes `BOmainPage.js` and `FOMainPage.js`. Of course, `CommonPage` is **not** meant to be instantiated (it’s kind of abstract class) !
+
+When you create new page classes, make sure to make them inherit from their respective `XXMainPage` to be sure to be able to use both generic methods and specific methods.
+
+Examples: 
+
+- Install page (which is independent) extends from `CommonPage`
+- Brands page (which is part of the BO) extends from `BOBasePage`
+- Checkout page (which is part of the FO) extends from `FOBasePage`  
+
+#### Methods
+Methods inside a page class must be properly named about their primary function. Remember to [KISS](https://en.wikipedia.org/wiki/KISS_principle) ! You should be able to understand what a method does just by looking at its name.
+
+Methods inside a page class **CANNOT USE** the `expect` keyword or any kind of asserting function, since their sole purpose is to interact with the page they’re linked with. They don’t validate anything per se, they just expose functionalities. 
+To validate a behavior in your test logic, your methods can return booleans, integers (number of lines, elements, etc), strings, objects... It’s up to you.
+
+Examples: 
+```
+   /**
+   * Get order status
+   * @return {Promise<string>}
+   */
+  async getOrderStatus() {
+    return this.getTextContent(`${this.orderStatusesSelect} option[selected='selected']`, false);
+  }
+```
+
+#### Selectors
+Selectors are used on every page and are stored as attributes of the class. They should be named following this convention:
+
+**nameType*** in `camelCase`.
+
+For example, a button used to submit the main form in the order page should be named: `submitMainFormButton`.
+
+
+#### Types
+Each selector must belong to a certain type. Here is a non-exhaustive list:
+- button
```

Comment:
You need to separate lists from the previous paragraph or else it won't work properly.

![Screenshot 2020-04-27 at 13 16 52](https://user-images.githubusercontent.com/1009343/80366321-63885e80-8889-11ea-9672-c84ff524d5b3.png)

I **strongly** suggest previewing your pages locally with Hugo (it's [very easy](https://github.com/PrestaShop/docs#rendering-the-site-locally)) before submitting them.

```suggestion

- button
```